lib/pbench/server/api/resources/users_api.py

```diff
@@ -510,25 +482,28 @@ def delete(self, username):
         # TODO: Check if the user has the right privileges
         if not verified and not user.is_admin():
             self.logger.warning(
-                "Username retrieved from the auth token {} is different from the username provided",
-                self.auth.auth.current_user().id,
+                "User {} is not authorized to delete user {}.", user.username, username,
             )
-            abort(403, message="Authentication token does not belong to the user")
+            abort(403, message="Not authorized to delete user")
 
         try:
-            if not User.delete(username):
-                self.logger.warning("Admin attempted to delete admin user")
+            user = User.query(username=username)
+            # Do not delete if the user is admin
+            if not user.is_admin():
+                User.delete(username)
         except Exception:
             self.logger.exception(
                 "Exception occurred during deleting the user entry for user '{}'",
                 username,
             )
             abort(500, message="INTERNAL ERROR")
         else:
+            if user.is_admin():
+                self.logger.warning("Admin attempted to delete admin user")
+                abort(405, message="Admin user can not be deleted")
```

Comment:
405 is an interesting choice (I would have opted for 403, but I can see using 405); however, a 405 response [requires](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/405):

> The server MUST generate an Allow header field in a 405 response containing a list of the target resource's currently supported methods.

Does Flask do that for us?

Reply:
And of course the `Allow` return wouldn't really be accurate or useful; the Users resource _does_ support `DELETE` but we've rejected it due to the nature of the requested resource. I agree `403` would be more appropriate.

Reply:
Fixed